repository/src/main/java/org/apache/atlas/discovery/EntityLineageService.java

```diff
@@ -459,12 +458,20 @@ private void traverseEdgesUsingBFS(String baseGuid, AtlasLineageListContext line
         Set<String> skippedVertices = new HashSet<>();
         Queue<String> traversalQueue = new LinkedList<>();
 
+        Map<String, List<String>> parentMapForNeighbours = new HashMap<>();  // New map to track parent nodes
```

Comment:
parentMapForNeighbours is a bit confusing, can rename to "lineageParentsForEntityMap" and "lineageChildrenForEntityMap"